### ISSUE TITLE
Native globbing on Unix platforms should pass syntactically-invalid-as-globbing-patterns unquoted literals through as-is

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#4971]
Related: #3931

An _unquoted_ token that is not a syntactically valid globbing pattern (wildcard expression) should be considered a _literal_ to be _passed as-is_ - not an _error_.

Note that, by default, `bash` retains whatever unquoted token you specify as-is, if it (a) happens not to be a valid globbing pattern or (b) happens not to match any files.
(Setting `shopt -s nullglob`, however, would result in syntactically valid non-matching globs evaluating to the _empty string_).

Steps to reproduce (Unix platforms)
------------------

```powershell
/bin/echo []
```

Expected behavior
-----------------

```none
[]
```

Actual behavior
---------------

```none
The specified wildcard character pattern is not valid: []
...
```

Environment data
----------------

```powershell
PowerShell Core v6.0.0-beta.7 on macOS 10.12.6
PowerShell Core v6.0.0-beta.7 on Ubuntu 16.04.3 LTS
PowerShell Core v6.0.0-beta.7 on Microsoft Windows 10 Pro (64-bit; v10.0.15063)
```

**Comment by SteveL-MSFT** [comment]
It seems the downside to making this change is if the user intended to use a valid pattern but typed it wrong

**Comment by mklement0** [comment]
Implementing implicit globbing for external utilities on Unix platforms is an attempt to _emulate_ the behavior of the _typical_ shell behavior there.

Defining _typical_ is tricky, as it depends on whether it refers to the default shell for _interactive_ use - which is often `bash` - vs. running _scripts_ via the POSIX-features-only `/bin/sh` shell (which is `bash` in  its POSIX-compatible mode on some platforms (e.g., macOS), and `dash` on others (e.g, Ubuntu), and possibly a different implementation on others).

Either way, the following - run on macOS 10.12.6 - shows that both `bash` and `dash` interpret unquoted `[]` as a _literal_ by default:

```sh 
$ shall -c '/bin/echo []
✓ sh (bash variant)                       [0.01s]
  []

✓ dash                                    [0.00s]
  []

✓ bash                                    [0.00s]
  []

✗ zsh                                     [0.02s]
  zsh:1: no matches found: []

✓ ksh                                     [0.02s]
  []

```

[`shall`](https://www.npmjs.com/package/shall) is a Unix utility of mine that runs a shell command against multiple POSIX-like shells.

**Comment by mklement0** [comment]
In other words: while you could argue that it's better for the shell to notify you of a malformed globbing pattern / expand a valid pattern that happens to match nothing to the empty string, this is _not_ how things work on Unix platforms by default - and emulating the latter is the goal, I presume.

**Comment by lzybkr** [comment]
Closed via #5188.

**Comment by iSazonov** [comment]
@SteveL-MSFT Can we remove Review-Committee review and add Fixed?
